packages/flutter/lib/src/widgets/interactive_viewer.dart

```diff
@@ -781,6 +781,12 @@ class _InteractiveViewerState extends State<InteractiveViewer> with TickerProvid
   // Handle an update to an ongoing gesture. All of pan, scale, and rotate are
   // handled with GestureDetector's scale gesture.
   void _onScaleUpdate(ScaleUpdateDetails details) {
+    widget.onInteractionUpdate?.call(ScaleUpdateDetails(
```

Comment:
Is there a reason a new `ScaleUpdateDetails` has to be constructed instead of just passing `details` to the callback?

Reply:
The [documentation](https://master-api.flutter.dev/flutter/widgets/InteractiveViewer/onInteractionUpdate.html) states:

> At the time this is called, the TransformationController will have already been updated to reflect the change caused by the interaction. 

Is the statement still true?

Reply:
About ScaleUpdateDetails: The reason was that I didn't want to pass things that weren't being applied, like independent vertical and horizontal scales.  However, given that I call this even when all interaction is disabled, I think you're right and I should just pass everything.

About the docs: You're right, thanks for catching that.  I think I should actually keep the behavior as it is documented.  I'll move my calls to onInteractionUpdate.